docs/design/ClassMatching.md

```diff
@@ -0,0 +1,398 @@
+- Extracts the required IDs from the classes using the Ramen labels on them
+- Compares the StorageIDs and required ReplicationIDs and updates status on
+  DRPolicy for entities that are common
+    - A new status field for the same would be defined
+      `status.[sync/async].peerClasses`
+    - It would be a list containing:
+
+    ```
+    - StorageClass: <value>
+      ReplicationID: <value> (optional)
+      StorageID:
+      - <value-C1>
+      - <value-C2>
+    ```
+
+    - The list would only contain SCs that either have a matched
+      Volume[Snapshot|Replication]Class across the clusters
+
+#### Example labels to peerClasses
+
+C1 and C2 are the peer clusters for workloads, Hub is the OCM hub cluster
+
+C1:
+
+- Contains the following classes and required labels
+    - Name: `SClass1`
+        - `ramendr.openshift.io/storageID: c1SID1`
+    - Name: `SClass2`
+        - `ramendr.openshift.io/storageID: c1SID2`
+    - Name: `VSClass1`
+        - `ramendr.openshift.io/storageID: c1SID1`
+    - Name: `VRClass1`
+        - `ramendr.openshift.io/storageID: c1SID1`
+        - `ramendr.openshift.io/replicationID: c1RID1`
+    - Name: `VSClass2`
+        - `ramendr.openshift.io/storageID: c1SID2`
+
+C2:
+
+- Is a mirror of C1, except SIDs have the c2 prefix (SID would be different
+  as these are different storage instances and refer to an async DR case)
+- Contains the following in addition:
+    - Name: `SClass3`
+        - `ramendr.openshift.io/storageID: c2SID3`
+    - Name: `SClass4`
+        - No labels
+
+Hub:
+
+- DRPolicy `status.async.peerClasses` (or VRG `spec.async.peerClasses`)
+
+```
+- StorageClass: SClass1
+  ReplicationID: c1RID1
+  StorageID:
+  - c1SID1
+  - c2SID1
+- StorageClass: SClass2
+  StorageID:
+  - c1SID1
+  - c2SID1
+```
+
+PVCs protected:
+
+- PVCs that use SClass1 or SClass2 will be protected by the appropriate VRG
+- PVCs that use SClass3 or SClass4 will report errors as there exists no
+  common replication scheme across the clusters for protection
+
+#### Determining DRPolicy as Sync/Async
+
+NOTE: This is a low priority requirement
+
+For each DRPolicy:
+
+- Compare ClassClaims for DRClusters in the policy to determine if SC names
+  and StorageID matches, resulting in a Sync policy
+- If they do not match, compare ClassClaims for DRClusters in the policy to
+  determine if ReplicationID matches, and/or there are VolumeSnapshotClasses
+  backing the StorageClasses, resulting in an Async policy
+    - If Async then require a drpolicy.spec.replicationInterval configured on
+      the DRPolicy
+
+NOTE: Currently the implementation just feeds off a drcluster.spec.region string
+that if equal between 2 DRClusters determines a Sync setup and if different
+determines an Async setup. This will not hold up when there are multiple storage
+instances on the ManagedCluster
+
+### DRPC reconciler changes to update VRG spec
+
+Based on the DRPolicy used by the DRPC, VRGs created will carry a supported
```

Comment:
We may potentially need `spec.peerClasses` than separate them out into `sync|async`.

Reply:
Reconsidered this, separation is fine though may be superfluous.